### ISSUE TITLE
Empty Response Body

### ISSUE THREAD

**Issue by atsharp** [MarcGiffing/bucket4j-spring-boot-starter#68]
I'm receiving empty response bodies when rate limiting occurs. I've verified that the configuration I'm using has the default content for rate limiting.
```
curl localhost:8977/test-object-count -v
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8977 (#0)
> GET /test-object-count HTTP/1.1
> Host: localhost:8977
> User-Agent: curl/7.64.1
> Accept: */*
> 
< HTTP/1.1 429 
< X-Rate-Limit-Retry-After-Seconds: 0
< Content-Type: application/json
< Content-Length: 0
< Date: Thu, 30 Jul 2020 16:23:17 GMT
< 
* Connection #0 to host localhost left intact
* Closing connection 0
```

The various headers seem to come through correctly, but as seen above there's no content.

I cloned the repo to test locally and found that adding `httpResponse.getWriter().flush();` after appending the response body [here](https://github.com/MarcGiffing/bucket4j-spring-boot-starter/blob/7294bee60c9a35df6ecc3d7ec5bac08a93db8d0d/bucket4j-spring-boot-starter/src/main/java/com/giffing/bucket4j/spring/boot/starter/servlet/ServletRequestFilter.java#L74) fixes the issue for me, but I'm unsure if that's an appropriate fix or if there is something else I'm missing.

Any advice would be greatly appreciated. Thanks!

**Comment by MarcGiffing** [comment]
Sorry for the really late response... 
Which Servlet container is in use? 
Did you tried it in your browser instead of curl?
Can somone else reproduce this problem?

**Comment by atsharp** [comment]
Let me try to put together an example repo

**Comment by atsharp** [comment]
To answer your earlier questions:
1) Using Embedded tomcat
2) The issue happens in multiple HTTP clients
3) I've managed to put a minimal repository demoing the problem (forked off of the Caffeine example project): https://github.com/atsharp/bucket4j-response-body-issue 
    `mvn clean test` to see the existing tests fail with an empty response body


This issue appears to happen when both Spring Security and [Logbook](https://github.com/zalando/logbook)(library which logs requests/responses) are enabled. From my experiments, disabling either of those libraries appears to resolve the issue. My guess is that in one or both of these libraries the servlet response is being wrapped and the content is being dropped before it's flushed by the servlet container.

If I had to guess, I'd say it's probably a bug in Logbook, though it's not obvious to me where it's happening. Likewise, as I stated in my initial report flushing the writer appears to resolve the issue, but not sure if that's a good practice.
